homekit/controller/controller.py

```diff
@@ -62,6 +63,49 @@ def discover(max_seconds=10):
         """
         return discover_homekit_devices(max_seconds)
 
+    @staticmethod
+    def discover_ble(max_seconds=10):
+        """
+        Perform a Bluetooth LE discovery for HomeKit accessory. It will listen for Bluetooth LE advertisement events
+        for the given amount of seconds. The result will be a list of dicts. The keys of the dicts are:
+
+         * name: the model name of the accessory (required)
+         * mac: the MAC address of the accessory (required)
+         * sf / flags: the numerical and human readable version of the status flags (supports pairing or not, see table
+                       6-32 page 125)
+         * device_id: the accessory's device id (required)
+         * acid / category: the category identifier in numerical and human readable form. For more information see table
+                            12-3 page 254 or homekit.Categories (required)
+         * gsn: Global State Number, increment on change of any characteristic, overflows at 65535.
+         * cn: the configuration number (required)
+         * cv: the compatible version
+
+        :param max_seconds: how long should the Bluetooth LE discovery should be performed (default 10s). See sleep for
+                            more details
+        :return: a list of dicts as described above
+        """
+        manager = HomekitDiscoveryDeviceManager('hci0')
+        manager.start_discovery()
+        DelayedExecution(manager.stop, max_seconds).start()
+        manager.run()
+        result = []
+        for discovered_device in manager.devices():
```

Comment:
Note the manager.devices() is not side effect free - it will call `update_devices()`. At best thats a waste of time making more dbus RPC calls over the wire.

Not a problem for your stuff right now, but It's worse if you are trying to run a discovering device manager in the background because the `Device` objects you have already discovered become invalid (or at least less valid). There will be a different instance of the Device class for the same mac, concurrently with the first one. So i've been trying to avoid update_devices() invocations.

Reply:
Ok understood, will add a method to our implementation of the device manager to skip `update_devices()`.